draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1293,12 +1302,18 @@ msg to a L-byte string, where L = ceil((ceil(log2(p)) + k) / 8); this
 string is then interpreted as an integer via OS2IP {{RFC8017}}. For example,
 for p a 255-bit prime and k = 128-bit security, L = ceil((255 + 128) / 8) = 48 bytes.
 
+Finally, hash\_to\_base sets the input to HKDF to H(msg) rather than to msg.
```

Comment:
My understanding is that the following are sufficient conditions for ensuring the usage of HKDF is indiferentiable from an RO when the hash function used by HMAC is modeled as an RO. These combine results from LBB19 (Lemma 8) and [[DRST12](https://eprint.iacr.org/2013/382.pdf)] (Theorems 4.3 and 4.4) See the discussion in [[PS19](https://eprint.iacr.org/2019/519.pdf), Appx. B] for details. Suppose that HMAC has output length `h` (in bits).

1. The "info" string is always length `u` (in bits).
2. The "salt" string is always length `h`.
3. The "key" string (i.e., `msg`) has length other than `u + 8` he length of the input to HMAC for the first HKDF-extracted block) or `h + u + 8` (the length of the input to HMAC for each subsequent block).

So `H(msg)` is suitable as long as `u !=  h - 8`.

By the way, what is `H`? Is the same hash function as used with HKDF?

Reply:
H is indeed the same hash used with HKDF. I'll double check that this is clear.

Thinking out loud here:

- HMAC's output length is the same as H's output length, which is `h` (we used `b` in the document, but I'll stick with your terminology here).

- `u` is 40 bits for us: `H2Cxy`, where "H2C" is ASCII text, and x and y are 8-bit values.

- As you say, input to the HMAC calls inside HKDF-Expand are either 48 or 48 + `h` bits, and so what we need is that `40 != h - 8` or, sufficiently, `h > 48`. But of course `h` will *definitely* be more than 48 because of the collision resistance requirements in this section.

So: I think we're in agreement here, and I think the text correctly states a sufficient requirement for `h`.

Reply:
(I've now added a cite to DRST12, too)

Reply:
Provided the sufficient conditions are correct, I agree with @kwantam's conclusion. (And thanks for writing that down, @cjpatton!)

Reply:
I would double check with Benjamin Lipp, just to be 100% we're interpreting his result correctly. (He's the lead author on the WireGuard analysis.)